sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient_extensions.cs

```diff
@@ -75,6 +76,52 @@ public WebPubSubServiceClient(Uri endpoint, string hub, AzureKeyCredential crede
             );
         }
 
+        /// <summary> Initializes a new instance of WebPubSubServiceClient. </summary>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="hub"> Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore. </param>
+        /// <param name="credential"> A token credential used to authenticate to an Azure Service. </param>
+        public WebPubSubServiceClient(Uri endpoint, string hub, TokenCredential credential)
+            : this(endpoint, hub, credential, new WebPubSubServiceClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of WebPubSubServiceClient. </summary>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="hub"> Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore. </param>
+        /// <param name="credential"> A token credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        public WebPubSubServiceClient(Uri endpoint, string hub, TokenCredential credential, WebPubSubServiceClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(hub, nameof(hub));
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            this._tokenCredential = credential;
+            this.hub = hub;
+            this.endpoint = endpoint;
+
+            options ??= new WebPubSubServiceClientOptions();
+            _clientDiagnostics = new ClientDiagnostics(options);
+            apiVersion = options.Version;
+
+            HttpPipelinePolicy[] perCallPolicies;
+            if (options.ReverseProxyEndpoint != null)
+            {
+                perCallPolicies = new HttpPipelinePolicy[] { new ApimPolicy(options.ReverseProxyEndpoint), new LowLevelCallbackPolicy() };
+            }
+            else
+            {
+                perCallPolicies = new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() };
+            }
+
+            Pipeline = HttpPipelineBuilder.Build(
+                options,
+                perCallPolicies: perCallPolicies,
+                perRetryPolicies: new HttpPipelinePolicy[] { new BearerTokenAuthenticationPolicy(credential, "TODO:credentialScopeName") },
```

Comment:
You want `"https://webpubsub.azure.com/.default"` for the scope.  I'd recommend making it a constant on your `ClientOptions` though so it's easier to find/update if we ever need to make this extensible.